marketplace-builder/src/builder.rs

```diff
@@ -332,54 +360,79 @@ mod test {
-                    }),
-            )
-            .network_config(network_config)
-            .build();
-        let _network = TestNetwork::new(config, MockSequencerVersions::new()).await;
+        let private_key =
+            <BLSPubKey as SignatureKey>::PrivateKey::generate(&mut rand::thread_rng());
+        let signature_key = BLSPubKey::from_private(&private_key);
+
+        signature_keys.push(signature_key);
+
+        // Initialize a rollup registration with namespace id = 10
+        let reg_ns_1_body = RollupRegistrationBody {
+            namespace_id: REGISTERED_NAMESPACE.into(),
+            reserve_url: Some(Url::from_str("http://localhost").unwrap()),
+            reserve_price: 200.into(),
+            active: true,
+            signature_keys: signature_keys.clone(),
+            text: "test".to_string(),
+            signature_key,
+        };
 
-        // Start the builder
-        let init = BuilderConfig::init(
-            true,
-            FeeAccount::test_key_pair(),
-            ViewNumber::genesis(),
-            NonZeroUsize::new(1024).unwrap(),
-            NonZeroUsize::new(1024).unwrap(),
-            NodeState::default(),
-            ValidatedState::default(),
-            event_service_url.clone(),
-            builder_api_url.clone(),
-            Duration::from_secs(2),
-            5,
-            Duration::from_secs(2),
-            FeeAmount::from(10),
-            Some(BidConfig {
-                namespaces: vec![NamespaceId::from(10u32)],
-                amount: FeeAmount::from(10),
-            }),
-            format!("http://localhost:{}", 3000).parse().unwrap(),
-        );
-        let _builder_config = init.await;
+        // Sign the registration body
+        let reg_signature = <SeqTypes as NodeType>::SignatureKey::sign(
+            &private_key,
+            reg_ns_1_body.commit().as_ref(),
+        )
+        .expect("failed to sign");
+
+        let reg_ns_1 = RollupRegistration {
+            body: reg_ns_1_body.clone(),
+            signature: reg_signature,
+        };
+
+        // registering a rollup
+        let _: RollupRegistration = client
+            .post("register_rollup")
+            .body_json(&reg_ns_1)
+            .unwrap()
+            .send()
+            .await
+            .unwrap();
+        let result: Vec<RollupRegistration> =
+            client.get("rollup_registrations").send().await.unwrap();
+        assert_eq!(result, vec![reg_ns_1]);
+
+        (mock_solver, solver_base_url)
+    }
 
+    /// Connect to the builder.
+    async fn connect_to_builder(
+        urls: QueryEventAndBuilderUrls,
+    ) -> Client<ServerError, MarketplaceVersion> {
         // Wait for at least one empty block to be sequenced (after consensus starts VID).
-        let sequencer_client: Client<ServerError, SequencerApiVersion> = Client::new(query_api_url);
+        let sequencer_client: Client<ServerError, SequencerApiVersion> = Client::new(urls.0);
```

Comment:
Super small nit: can we add a comment explaining what urls.0 and urls.2 represent?  I know it's explained above in the type definition, but it would be helpful here as well.

Reply:
Or perhaps we can just define each of the 3 urls as their own type with a descriptive name?

Reply:
I'll give each URL a name, good suggestion!